src/utils/components/SSHSecretSection/utils/components/SelectSecret.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, ReactElement, useState } from 'react';
+
+import { SecretModel } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Secret } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { validateSSHPublicKey } from '@kubevirt-utils/utils/utils';
+import { ResourceLink, WatchK8sResult } from '@openshift-console/dynamic-plugin-sdk';
+import { Alert, AlertVariant, Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import Loading from '../../../Loading/Loading';
+import { decodeSecret } from '../utils';
+
+type SelectSecretProps = {
+  secretsResourceData: WatchK8sResult<IoK8sApiCoreV1Secret[]>;
+  selectedSecretName: string;
+  onSelectSecret: (secretName: string) => void;
+  id?: string;
+};
+
+const SelectSecret: FC<SelectSecretProps> = ({
+  secretsResourceData,
+  selectedSecretName,
+  onSelectSecret,
+  id,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isSecretSelectOpen, setSecretSelectOpen] = useState(false);
+
+  const [secrets, secretsLoaded, secretsError] = secretsResourceData;
+  const sshKeySecrets = secrets
+    ? secrets?.filter((secret) => validateSSHPublicKey(decodeSecret(secret)))
+    : [];
+
+  const filterSecrets = (_, value: string): ReactElement[] => {
+    let filteredSecrets = sshKeySecrets;
+
+    if (value) {
+      const regex = new RegExp(value, 'i');
+      filteredSecrets = sshKeySecrets.filter((secret) => regex.test(secret?.metadata?.name));
+    }
+
+    return filteredSecrets.map((secret) => (
+      <SelectOption key={secret?.metadata?.name} value={secret?.metadata?.name}>
+        <ResourceLink kind={SecretModel.kind} name={secret?.metadata?.name} linkTo={false} />
+      </SelectOption>
+    )) as ReactElement[];
+  };
```

Comment:
```suggestion
  const filterSecrets = (_, value = '': string): ReactElement[] => {
    const regex = new RegExp(value, 'i');
    filteredSecrets = sshKeySecrets.filter((secret) => regex.test(secret?.metadata?.name));

   return filteredSecrets.map((secret) => (
     <SelectOption key={secret?.metadata?.name} value={secret?.metadata?.name}>
       <ResourceLink kind={SecretModel.kind} name={secret?.metadata?.name} linkTo={false} />
     </SelectOption>
   )) as ReactElement[];
 };
```

What I mean is to check if something like this should work.
Also, I would have split this function, as this function eventually returns elements and is not related to filterSecret meaning. and the actually filter function can be in a util file

Reply:
Ugo added the if statement to avoid iterating over the array if there was no input, but you're correct that it isn't necessary since the filter function isn't called unless there's input. I removed it.

I don't believe I can move it to a util file because it needs `sshKeySecrets` and as far as I can tell there's no way to pass the list of options to the function as a parameter.